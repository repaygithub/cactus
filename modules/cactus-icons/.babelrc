{
  "presets": ["@repay/babel-preset"],
  "sourceMaps": "inline",
  "plugins": [["@babel/plugin-proposal-private-property-in-object", { "loose": true }]]
}