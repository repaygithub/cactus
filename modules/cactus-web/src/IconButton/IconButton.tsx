import React from 'react'
import styled, { FlattenInterpolation, ThemeProps, css } from 'styled-components'
import { space, SpaceProps } from 'styled-system'
import { CactusTheme } from '@repay/cactus-theme'
import { Omit } from '../types'
import splitProps from '../helpers/splitProps'

export type IconButtonVariants = 'standard' | 'action'
export type IconButtonSizes = 'tiny' | 'small' | 'medium' | 'large'

interface IconButtonProps
  extends Omit<
      React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>,
      'ref'
    >,
    SpaceProps {
  iconSize?: IconButtonSizes
  variant?: IconButtonVariants
  disabled?: boolean
  label?: string
  display?: 'flex' | 'inline-flex'
  inverse?: boolean
}

type SizeMap = { [K in IconButtonSizes]: string }

const sizeMap: SizeMap = {
  tiny: '8px',
  small: '16px',
  medium: '24px',
  large: '40px',
}

type VariantMap = { [K in IconButtonVariants]: FlattenInterpolation<ThemeProps<CactusTheme>> }

const variantMap: VariantMap = {
  action: css`
    color: ${p => p.theme.colors.callToAction};

    &:hover,
    &:focus {
      color: ${p => p.theme.colors.base};
    }
  `,
  standard: css`
    color: ${p => p.theme.colors.base};

    &:hover,
    &:focus {
      color: ${p => p.theme.colors.callToAction};
    }
  `,
}

const inverseVariantMap: VariantMap = {
  action: css`
    color: ${p => p.theme.colors.callToAction};

    &:hover,
    &:focus {
      color: ${p => p.theme.colors.white};
    }
  `,
  standard: css`
    color: ${p => p.theme.colors.white};

    &:hover,
    &:focus {
      color: ${p => p.theme.colors.base};
      background: ${p => p.theme.colors.white};
    }
  `,
}

const disabled: FlattenInterpolation<ThemeProps<CactusTheme>> = css`
  color: ${p => p.theme.colors.mediumGray};
  cursor: not-allowed;
`

const variantOrDisabled = (
  props: IconButtonProps
): FlattenInterpolation<ThemeProps<CactusTheme>> | undefined => {
  const map = props.inverse ? inverseVariantMap : variantMap
  if (props.disabled) {
    return disabled
  } else if (props.variant !== undefined) {
    return map[props.variant]
  }
}

const IconButtonBase = React.forwardRef<HTMLButtonElement, IconButtonProps>((props, ref) => {
  const [componentProps, marginProps] = splitProps<IconButtonProps>(props, 'IconButton')
  const { label, children, inverse, iconSize, ...buttonProps } = componentProps

  return (
    <button aria-label={label} ref={ref} {...buttonProps} {...marginProps}>
      {children}
    </button>
  )
})

export const StyledIconButton = styled.button<IconButtonProps>`
  display: ${p => p.display || 'inline-flex'};
  font-size: ${p => p.iconSize !== undefined && sizeMap[p.iconSize]};
  align-items: center;
  justify-content: center;
  padding: 0px;
  border-radius: 50%;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;

  ${space}
  ${variantOrDisabled}
`

const IconButton = (props: IconButtonProps) => {
  const [componentProps, marginProps] = splitProps<IconButtonProps>(props, 'IconButton')
  const { label, children } = componentProps

  return (
    <StyledIconButton aria-label={label} {...componentProps} {...marginProps}>
      {children}
    </StyledIconButton>
  )
}

IconButton.defaultProps = {
  variant: 'standard',
  iconSize: 'medium',
  disabled: false,
  inverse: false,
}

export default IconButton
