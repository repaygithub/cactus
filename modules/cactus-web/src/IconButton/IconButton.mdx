---
name: IconButton
menu: Components
---

import PropsTable from 'website-src/components/PropsTable'
import { livePreviewStyle } from '../helpers/constants'
import IconButton from './IconButton'
import cactusTheme from '@repay/cactus-theme'
import { ActionsAdd } from '@repay/cactus-icons'
import { LiveProvider, LiveEditor, LiveError, LivePreview } from 'react-live'

# IconButton

The `IconButton` allows you to turn icons into buttons, and it was designed to allow the use of icons from any library.

### Try it out

export const code = `<IconButton label="add">
  <ActionsAdd />
</IconButton>`

<LiveProvider code={code} scope={{ IconButton, ActionsAdd }}>
  <LiveEditor style={livePreviewStyle} />
  <LiveError />
  <LivePreview />
</LiveProvider>

## Best practices

- To ensure that the `IconButton` is accessible, use a descriptive `label`. This will set the `aria-label` on the button.
- Icon sizes should not be defined manually; the size should be defined using the `iconSize` prop.

## Basic usage

```jsx
import { IconButton } from '@repay/cactus-web'
import { StatusCheck } from '@repay/cactus-icons'
...
// Basic usage
<IconButton>
  <StatusCheck />
</IconButton>

// Call to action variant
<IconButton variant="action">
  <StatusCheck />
</IconButton>

// Inverse
<IconButton inverse={true}>
  <StatusCheck />
</IconButton>

// Sizing
<IconButton iconSize="large">
  <StatusCheck />
</IconButton>

// Disabled
<IconButton disabled={true}>
  <StatusCheck />
</IconButton>
```

## Properties

<PropsTable of={IconButton} />
