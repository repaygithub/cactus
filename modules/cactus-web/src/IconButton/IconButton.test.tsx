import * as React from 'react'

import { cleanup, fireEvent, render } from '@testing-library/react'
import { StatusCheck } from '@repay/cactus-icons'
import { StyleProvider } from '../StyleProvider/StyleProvider'
import cactusTheme from '@repay/cactus-theme'
import IconButton from './IconButton'

afterEach(cleanup)

describe('component: IconButton', () => {
  test('should default to standard medium icon button', () => {
    const iconButton = render(
      <StyleProvider>
        <IconButton label="boolest">
          <StatusCheck />
        </IconButton>
      </StyleProvider>
    )

    expect(iconButton.asFragment()).toMatchSnapshot()
  })

  test('should render standard medium icon button', () => {
    const iconButton = render(
      <StyleProvider>
        <IconButton label="boolest" iconSize="medium">
          <StatusCheck />
        </IconButton>
      </StyleProvider>
    )

    expect(iconButton.asFragment()).toMatchSnapshot()
  })

  test('should render tiny icon button', () => {
    const iconButton = render(
      <StyleProvider>
        <IconButton label="boolest" iconSize="tiny">
          <StatusCheck />
        </IconButton>
      </StyleProvider>
    )

    expect(iconButton.asFragment()).toMatchSnapshot()
  })

  test('should render small icon button', () => {
    const iconButton = render(
      <StyleProvider>
        <IconButton label="boolest" iconSize="small">
          <StatusCheck />
        </IconButton>
      </StyleProvider>
    )

    expect(iconButton.asFragment()).toMatchSnapshot()
  })

  test('should render large icon button', () => {
    const iconButton = render(
      <StyleProvider>
        <IconButton label="boolest" iconSize="large">
          <StatusCheck />
        </IconButton>
      </StyleProvider>
    )

    expect(iconButton.asFragment()).toMatchSnapshot()
  })

  test('should render call to action icon button', () => {
    const iconButton = render(
      <StyleProvider>
        <IconButton label="boolest" variant="action">
          <StatusCheck />
        </IconButton>
      </StyleProvider>
    )

    expect(iconButton.asFragment()).toMatchSnapshot()
  })

  test('should render danger icon button', () => {
    const iconButton = render(
      <StyleProvider>
        <IconButton label="boolest" variant="danger">
          <StatusCheck />
        </IconButton>
      </StyleProvider>
    )

    expect(iconButton.asFragment()).toMatchSnapshot()
  })

  test('should render inverse standard icon button', () => {
    const iconButton = render(
      <StyleProvider>
        <IconButton label="boolest" variant="standard" inverse>
          <StatusCheck />
        </IconButton>
      </StyleProvider>
    )

    expect(iconButton.asFragment()).toMatchSnapshot()
  })

  test('should render inverse call to action icon button', () => {
    const iconButton = render(
      <StyleProvider>
        <IconButton label="boolest" variant="action" inverse>
          <StatusCheck />
        </IconButton>
      </StyleProvider>
    )

    expect(iconButton.asFragment()).toMatchSnapshot()
  })

  test('should render inverse danger icon button', () => {
    const iconButton = render(
      <StyleProvider>
        <IconButton label="boolest" variant="danger" inverse>
          <StatusCheck />
        </IconButton>
      </StyleProvider>
    )

    expect(iconButton.asFragment()).toMatchSnapshot()
  })

  test('should render icon button with aria-label', () => {
    const iconButton = render(
      <StyleProvider>
        <IconButton label="uchiha-itachi">
          <StatusCheck />
        </IconButton>
      </StyleProvider>
    )

    expect(iconButton.asFragment()).toMatchSnapshot()
  })

  test('should render disabled variant', () => {
    const iconButton = render(
      <StyleProvider>
        <IconButton label="boolest" disabled>
          <StatusCheck />
        </IconButton>
      </StyleProvider>
    )

    expect(iconButton.asFragment()).toMatchSnapshot()
  })

  test('should render inverse disabled variant', () => {
    const iconButton = render(
      <StyleProvider>
        <IconButton label="boolest" inverse disabled>
          <StatusCheck />
        </IconButton>
      </StyleProvider>
    )

    expect(iconButton.asFragment()).toMatchSnapshot()
  })

  test('should support margin space props', () => {
    const iconButton = render(
      <StyleProvider>
        <IconButton label="boolest" mb={4}>
          <StatusCheck />
        </IconButton>
      </StyleProvider>
    )

    expect(iconButton.asFragment()).toMatchSnapshot()
  })

  test('should trigger onClick', () => {
    const onClick = jest.fn()
    const { getByTestId } = render(
      <StyleProvider>
        <IconButton label="boolest" onClick={onClick} data-testid="clicked">
          <StatusCheck />
        </IconButton>
      </StyleProvider>
    )

    fireEvent.click(getByTestId('clicked'))
    expect(onClick).toHaveBeenCalled()
  })

  test('should not trigger onClick', () => {
    const onClick = jest.fn()
    const { getByTestId } = render(
      <StyleProvider>
        <IconButton label="boolest" disabled onClick={onClick} data-testid="not-clicked">
          <StatusCheck />
        </IconButton>
      </StyleProvider>
    )

    fireEvent.click(getByTestId('not-clicked'))
    expect(onClick).not.toHaveBeenCalled()
  })
})
