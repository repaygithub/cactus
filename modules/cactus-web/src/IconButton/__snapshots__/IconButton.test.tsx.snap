// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: IconButton should default to standard medium icon button 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  border-radius: 50%;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
  color: hsl(200,96%,11%);
}

.c0:hover,
.c0:focus {
  color: hsl(200,96%,35%);
}

.c1 {
  vertical-align: middle;
  width: 1em;
}

<button
    class="c0"
  >
    <svg
      class="c1"
      fill="currentcolor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`component: IconButton should not support padding space props 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  border-radius: 50%;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
  color: hsl(200,96%,11%);
}

.c0:hover,
.c0:focus {
  color: hsl(200,96%,35%);
}

.c1 {
  vertical-align: middle;
  width: 1em;
}

<button
    class="c0"
  >
    <svg
      class="c1"
      fill="currentcolor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`component: IconButton should render call to action icon button 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  border-radius: 50%;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
  color: hsl(200,96%,35%);
}

.c0:hover,
.c0:focus {
  color: hsl(200,96%,11%);
}

.c1 {
  vertical-align: middle;
  width: 1em;
}

<button
    class="c0"
  >
    <svg
      class="c1"
      fill="currentcolor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`component: IconButton should render disabled variant 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  border-radius: 50%;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
  color: hsl(0,0%,70%);
  cursor: not-allowed;
}

.c1 {
  vertical-align: middle;
  width: 1em;
}

<button
    class="c0"
    disabled=""
  >
    <svg
      class="c1"
      fill="currentcolor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`component: IconButton should render icon button with aria-label 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  border-radius: 50%;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
  color: hsl(200,96%,11%);
}

.c0:hover,
.c0:focus {
  color: hsl(200,96%,35%);
}

.c1 {
  vertical-align: middle;
  width: 1em;
}

<button
    aria-label="uchiha-itachi"
    class="c0"
    label="uchiha-itachi"
  >
    <svg
      class="c1"
      fill="currentcolor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`component: IconButton should render inverse call to action icon button 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  border-radius: 50%;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
  color: hsl(200,96%,35%);
}

.c0:hover,
.c0:focus {
  color: hsl(0,0%,100%);
}

.c1 {
  vertical-align: middle;
  width: 1em;
}

<button
    class="c0"
  >
    <svg
      class="c1"
      fill="currentcolor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`component: IconButton should render inverse disabled variant 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  border-radius: 50%;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
  color: hsl(0,0%,70%);
  cursor: not-allowed;
}

.c1 {
  vertical-align: middle;
  width: 1em;
}

<button
    class="c0"
    disabled=""
  >
    <svg
      class="c1"
      fill="currentcolor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`component: IconButton should render inverse standard icon button 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  border-radius: 50%;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
  color: hsl(0,0%,100%);
}

.c0:hover,
.c0:focus {
  color: hsl(200,96%,11%);
  background: hsl(0,0%,100%);
}

.c1 {
  vertical-align: middle;
  width: 1em;
}

<button
    class="c0"
  >
    <svg
      class="c1"
      fill="currentcolor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`component: IconButton should render large icon button 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 40px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  border-radius: 50%;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
  color: hsl(200,96%,11%);
}

.c0:hover,
.c0:focus {
  color: hsl(200,96%,35%);
}

.c1 {
  vertical-align: middle;
  width: 1em;
}

<button
    class="c0"
  >
    <svg
      class="c1"
      fill="currentcolor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`component: IconButton should render small icon button 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  border-radius: 50%;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
  color: hsl(200,96%,11%);
}

.c0:hover,
.c0:focus {
  color: hsl(200,96%,35%);
}

.c1 {
  vertical-align: middle;
  width: 1em;
}

<button
    class="c0"
  >
    <svg
      class="c1"
      fill="currentcolor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`component: IconButton should render standard medium icon button 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  border-radius: 50%;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
  color: hsl(200,96%,11%);
}

.c0:hover,
.c0:focus {
  color: hsl(200,96%,35%);
}

.c1 {
  vertical-align: middle;
  width: 1em;
}

<button
    class="c0"
  >
    <svg
      class="c1"
      fill="currentcolor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`component: IconButton should render tiny icon button 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  border-radius: 50%;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
  color: hsl(200,96%,11%);
}

.c0:hover,
.c0:focus {
  color: hsl(200,96%,35%);
}

.c1 {
  vertical-align: middle;
  width: 1em;
}

<button
    class="c0"
  >
    <svg
      class="c1"
      fill="currentcolor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`component: IconButton should support margin space props 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px;
  border-radius: 50%;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
  margin-bottom: 16px;
  color: hsl(200,96%,11%);
}

.c0:hover,
.c0:focus {
  color: hsl(200,96%,35%);
}

.c1 {
  vertical-align: middle;
  width: 1em;
}

<button
    class="c0"
  >
    <svg
      class="c1"
      fill="currentcolor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
    >
      <path
        d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
      />
    </svg>
  </button>
</DocumentFragment>
`;
